### ISSUE TITLE
pex can pull in a bad pkg_resources

### ISSUE THREAD

**Issue by wickman** [pantsbuild/pex#54]
There are two non-namespace packages on pypi that share the 'pants' namespace: pants and pantsbuild.pants.  If they're both installed in your environment or virtualenv, importing pkg_resources will stacktrace.  This means that pex won't work in that situation.

I'm not sure which version of pkg_resources is broken in particular, but I know at least the one bundled in setuptools 12.2 does not suffer the following problem, so we should pin pex to something sufficiently new to avoid this:

```shell

# vendored pkg_resources
(ve)mba=ve=; PYTHONPATH=/Users/wickman/.tools-cache/home/mesos/tools/client/libexec/aurora-client/.bootstrap python
Python 2.7.8 (default, Nov  3 2014, 10:09:47) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pkg_resources
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/wickman/.tools-cache/home/mesos/tools/client/libexec/aurora-client/.bootstrap/pkg_resources.py", line 2736, in <module>
    add_activation_listener(lambda dist: dist.activate())
  File "/Users/wickman/.tools-cache/home/mesos/tools/client/libexec/aurora-client/.bootstrap/pkg_resources.py", line 698, in subscribe
    callback(dist)
  File "/Users/wickman/.tools-cache/home/mesos/tools/client/libexec/aurora-client/.bootstrap/pkg_resources.py", line 2736, in <lambda>
    add_activation_listener(lambda dist: dist.activate())
  File "/Users/wickman/.tools-cache/home/mesos/tools/client/libexec/aurora-client/.bootstrap/pkg_resources.py", line 2274, in activate
    list(map(declare_namespace, self._get_metadata('namespace_packages.txt')))
  File "/Users/wickman/.tools-cache/home/mesos/tools/client/libexec/aurora-client/.bootstrap/pkg_resources.py", line 1882, in declare_namespace
    _handle_ns(packageName, path_item)
  File "/Users/wickman/.tools-cache/home/mesos/tools/client/libexec/aurora-client/.bootstrap/pkg_resources.py", line 1849, in _handle_ns
    loader.load_module(packageName)
  File "/Users/wickman/Python/CPython-2.7.8/lib/python2.7/pkgutil.py", line 246, in load_module
    mod = imp.load_module(fullname, self.file, self.filename, self.etc)
  File "/private/tmp/ve/lib/python2.7/site-packages/pants/__init__.py", line 31, in <module>
    from pants._channel import HAS_IPV6, HAS_UNIX
  File "/private/tmp/ve/lib/python2.7/site-packages/pants/_channel.py", line 34, in <module>
    from pants.engine import Engine
ImportError: cannot import name Engine
>>> 
(ve)mba=ve=; 


# new pkg_resources
(ve)mba=ve=; python
Python 2.7.8 (default, Nov  3 2014, 10:09:47) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pkg_resources
>>> 
```

**Comment by jsirois** [comment]
I think its valid to say pantsbuild.pants is simply at fault here.  It used the name since pants was taken and it correspondingly should have used a non-taken namespace.  I'm happy to raise this on pants-devel if you agree and push for a pantsbuild.pants global package re-nesting.

**Comment by wickman** [comment]
Going to mark as fixed as of c4b6e0f because one at least has the option to pin to something >= 12.2.
